app/src/main/java/ar/com/wolox/android/cookbook/scanqr/captureqr/CaptureQrActivity.kt

```diff
@@ -0,0 +1,99 @@
+package ar.com.wolox.android.cookbook.scanqr.captureqr
+
+import android.content.Intent
+import android.content.pm.PackageManager
+import android.os.Bundle
+import android.view.KeyEvent
+import ar.com.wolox.android.cookbook.R
+import ar.com.wolox.android.cookbook.scanqr.ScanQrActivity
+import ar.com.wolox.wolmo.core.activity.WolmoActivity
+import com.journeyapps.barcodescanner.CaptureManager
+import com.journeyapps.barcodescanner.DecoratedBarcodeView
+import com.journeyapps.barcodescanner.ViewfinderView
+import kotlinx.android.synthetic.main.custom_qr_scanner.*
+import kotlinx.android.synthetic.main.fragment_capture_qr.*
+import javax.inject.Inject
+
+class CaptureQrActivity : WolmoActivity(), DecoratedBarcodeView.TorchListener, CaptureQrView {
+    @Inject
+    lateinit var presenter: CaptureQrPresenter
+
+    private var capture: CaptureManager? = null
+    private var barcodeScannerView: DecoratedBarcodeView? = null
+    private var viewfinderView: ViewfinderView? = null
+
+    override fun layout(): Int = R.layout.fragment_capture_qr
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setUpScannerView(savedInstanceState)
```

Comment:
Same as above, presenter should invoke this.

Reply:
(already discussed this)

Reply:
Just to left here what we talked in private:
- The scanner library needs the savedInstanceState and the presenter shouldn't know a Bundle, so this should remain here.
- It would be great to have a `Scanner` or `ScannerAdapter` class responsible of scanner handle, to avoid putting this logic on the fragment, but it won't be prioritised for the moment.